.travis.yml

```diff
@@ -28,17 +30,24 @@ addons:
       - llvm-toolchain-precise
 
 env:
-  - COMPILER=clang++-3.6 CMAKE_GENERATOR="Ninja"
-  - COMPILER=clang++-3.6 CMAKE_GENERATOR="Unix Makefiles"
-
-  - COMPILER=clang++-3.7 CMAKE_GENERATOR="Unix Makefiles"
-  - COMPILER=clang++-3.7 CMAKE_GENERATOR="Ninja"
-
-  - COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
-  - COMPILER=clang++-3.8 CMAKE_GENERATOR="Ninja"
-
-  - COMPILER=gcc CMAKE_GENERATOR="Ninja"
-  - COMPILER=gcc CMAKE_GENERATOR="Unix Makefiles"
+  # Test all generators and Clang + GCC
+  - TEST=true COMPILER=clang++-3.6 CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=clang++-3.6 CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=clang++-3.7 CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=clang++-3.7 CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=g++-4.8     CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=g++-4.8     CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=g++-4.9     CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=g++-4.9     CMAKE_GENERATOR="Unix Makefiles"
```

Comment:
I was apprehending that GCC 5 would report itself as C++14 compliant despite its support for C++14 being completely broken, which would break the benchmarks for Hana. I thought it would be best tackled in a subsequent PR.